python/ray/dag/output_node.py

```diff
@@ -0,0 +1,47 @@
+import ray
+from typing import Any, Dict, List, Union, Tuple
+
+from ray.dag import DAGNode
+from ray.dag.format_utils import get_dag_node_str
+
+IN_CONTEXT_MANAGER = "__in_context_manager__"
+
+
+class OutputNode(DAGNode):
```

Comment:
Is it only useful when there are multiple outputs? If so, how about calling it something like MultiOutputNode?

Reply:
actually I thought we want this even for a single output (since we need to allocate the buffer ahead of time). cc @stephanie-wang can we allocate the output buffer without this Output node? for an single output case?

Reply:
For single-output, it's kind of a nuisance to require users to specify it. We don't really need it for compiled DAG; it's just useful to know which node is the sink. My plan for compiled DAG was to add it implicitly if the user didn't specify it.

Reply:
So yes I agree that we can just call it MultiOutputNode, since users only need to use if there are multiple outputs.